suzieq/poller/worker/nodes/node.py

```diff
@@ -463,6 +463,10 @@ async def _detect_node_type(self):
                 # that sets the device type
                 devtype = self.devtype
             except Exception:
+                self.logger.exception('Node discovery failed due to exception')
+                # As there was an exception there is not point in retrying
+                # discovery, it is likely a bug
+                self._retry = 0
```

Comment:
This exception is not Timeout or any such known error, correct? Maybe adding that to the comment would help.